src/nvim/memory.c

```diff
@@ -370,6 +370,34 @@ size_t xstrlcpy(char *restrict dst, const char *restrict src, size_t size)
     return ret;
 }
 
+/// xstrlcat - appends string src to the end of dst.
+///
+/// Compatible with *BSD strlcat: It will append at most
+/// dstsize - strlen(dst) - 1 characters.
+/// It will then NUL-terminate, unless dstsize is 0 or
+/// the original dst string was longer than dstsize
+/// (in practice this should not happen as it means
+/// that either dstsize is incorrect or that dst is not a proper string).
+///
+/// @param dst Where to copy the string to
+/// @param src Where to copy the string from
+/// @param size Size of destination buffer
+/// @return the initial length of dst plus the length of src.
+///         (i.e. strlen(src) + strlen(dst))
+size_t xstrlcat(char *restrict dst, const char *restrict src, size_t dstsize)
+  FUNC_ATTR_NONNULL_ALL
+{
+    size_t src_len = strlen(src);
+    size_t dst_len = strlen(dst);
+    size_t ret = src_len + dst_len;
+    if (src_len) {
+        size_t len = (ret >= dstsize) ? dstsize - 1 : ret;
```

Comment:
In the function comment you said that `dstsize` *may* be zero. But this situation is not handled: in this case `len = dstsize - 1 = (size_t) -1`,  `len - dst_len = (size_t) -1` which means overrun with *any* size of `dst` and `src`. Since handling `dstsize == 0` does not make much sense I think it is better to update comment.